pysb/generator/kappa.py

```diff
@@ -51,56 +62,88 @@ def generate_molecule_types(self):
         for obs in self.model.observables:
             name = '\'' + obs.name + '\''
             observable_code = format_reactionpattern(obs.reaction_pattern)
+            # In the near future (KaSim 4.0), the observable syntax will
+            # require pipe characters around the expression. However, for the
+            # time being we'll stick with the old syntax for backwards
+            # compatibility
+            #if self.dialect == 'kasim':
+            #    self.__content += ("%%obs: %s |%s|\n") % \
+            #                      (name, observable_code)
+            #else:
+            #    self.__content += ("%%obs: %s %s\n") % (name, observable_code)
             self.__content += ("%%obs: %s %s\n") % (name, observable_code)
+
         self.__content += "\n"
-        #self.__content += "end observables\n\n"
 
     def generate_species(self):
         if not self.model.initial_conditions:
-            raise Exception("BNG generator requires initial conditions.")
-        species_codes = [format_complexpattern(cp) for cp, param in self.model.initial_conditions]
+            warnings.warn("Warning: No initial conditions.")
+
+        species_codes = [format_complexpattern(cp)
+                         for cp, param in self.model.initial_conditions]
         #max_length = max(len(code) for code in species_codes)
-        max_length = 0
-        #self.__content += "begin species\n"
         for i, code in enumerate(species_codes):
             param = self.model.initial_conditions[i][1]
-            #self.__content += ("%%init:  %-" + str(max_length) + "s   %s\n") % (code, param.name)
+            #self.__content += ("%%init:  %-" + str(max_length) + \
+            #                  "s   %s\n") % (code, param.name)
             if (self.dialect == 'kasim'):
-                # Switched from %g (float) to %d (int) because kappa didn't like scientific notation
-                # for large integers
-                self.__content += ("%%init: %d %s\n") % (float(param.value), code)
+                # Switched from %g (float) to %d (int) because kappa didn't
+                # like scientific notation for large integers
+                self.__content += ("%%init: \'%s\' %s\n") % (param.name, code)
                 #self.__content += ("%%init: %10g %s\n") % (param.value, code)
             else:
-                # Switched from %g (float) to %d (int) because kappa didn't like scientific notation
-                # for large integers
-                self.__content += ("%%init: %10d * %s\n") % (float(param.value), code)
+                if isinstance(param,pysb.core.Expression):
+                    raise KappaException("complx does not support Expressions.")
+                # Switched from %g (float) to %d (int) because kappa didn't
+                # like scientific notation for large integers
+                self.__content += ("%%init: %10d * %s\n") % \
+                                  (float(param.value), code)
```

Comment:
Why is param.value being converted to an float before being formatted as an integer? Can we just pass param.value as it is?